### DISCUSSION TITLE
Invalid timezone on Windows Subsystem for Linux (WSL)

### DISCUSSION THREAD

**Discussion by JetGum** [datenstrom/community#788] [Report a bug] [answered]
Hi,

I just did a clean install of latest Yellow, startet built-in web server. Worked fine.
Then I wanted to install extensions via CLI, and got this error message:

```
PHP Notice:  date_default_timezone_set(): Timezone ID 'Europe' is invalid in core.php on line 95
ERROR installing files: Can't connect to the update server!
```
Turns out that in yellow-system.ini, `"CoreTimezone: Europe"` had to be changed to `CoreTimezone: UTC`.

Cheers,

Carsten

**Comment by schulle4u** [comment]
Yellow uses `UTC` by default and if no timezone has been set in your php configuration. Please check your webhosts php timezone setting. [A valid timezone](https://www.php.net/manual/timezones.php) usually has the format continent/city, e.g Europe/Berlin.

**Comment by annaesvensson** [comment]
What does the log say? Can you reproduce this every time in a new install?

**Reply by JetGum** [reply]
Yes, I just reproduced it with a clean install (small blog) on my local machine.

1. unzip latest Yellow in to a folder
2. via CLI (Visual Studio Code Terminal): `php yellow.php serve`
3. via frontend: call localhost:8000 and installed a "small blog"
4. via CLI:  `php yellow.php update`
5. yellow-website.log says: `2022-05-26 10:07:36 error Can't parse file 'system/extensions/core.php'!`

Changing yellow-system.ini settings to `CoreTimezone: Europe/Berlin` or `CoreTimezone: UTC` fixes the problem.

Maybe (as Steffen suggested) this could be a problem with my PHP settings on localhost? I'm running a Win10, running a virtual machine WSL Ubuntu 20.4). But I do not know how to check the phpinfo-settings in these surrounding, sorry :-)

**Reply by annaesvensson** [reply]
Please copy/paste the full log.

**Reply by JetGum** [reply]
2022-05-26 08:07:15 info Install Datenstrom Yellow 0.8.20, PHP 7.4.3, Built-in 7.4.3, Linux
2022-05-26 08:07:15 info Install extension 'English 0.8.32'
2022-05-26 08:07:15 info Install extension 'German 0.8.32'
2022-05-26 08:07:15 info Install extension 'Swedish 0.8.32'
2022-05-26 08:07:32 info Install extension 'Blog 0.8.19'
2022-05-26 08:07:32 info Add user 'Admin'
2022-05-26 10:07:36 error Can't parse file 'system/extensions/core.php'!

**Reply by annaesvensson** [reply]
Thank you, this is what we needed to know to narrow down the cause of the problem.

**Comment by schulle4u** [comment]
@JetGum Please run this command in your WSL environment: 

`sudo dpkg-recoonfigure tzdata` 

Select Europe from the first menu, then a city near your location. This should fix your invalid timezone setting. 

Steffen

**Reply by JetGum** [reply]
Hi Steffen, 

I was able to choose Europe/Berlin as timezone in my WSL env.

Thanks for your support!

**Comment by schulle4u** [comment, marked as answer]
An update to the core's timezone detection has been published which fixes invalid system timezone settings during installation. If a misconfigured timezone is detected, yellow's timezone will be set to UTC by default.
